src/overmap.cpp

```diff
@@ -4364,6 +4454,424 @@ void overmap::populate_connections_out_from_neighbors( const overmap *north, con
     } );
 }
 
+void overmap::place_highways()
+{
+    const point_abs_om this_om = pos();
+    const int this_om_x = this_om.x();
+    const int this_om_y = this_om.y();
+    bool ocean_next_north = false;
+    bool ocean_next_east = false;
+    bool ocean_next_south = false;
+    bool ocean_next_west = false;
+    const city &invalid_city = get_invalid_city();
+    if( get_option<bool>( "OVERMAP_PLACE_OCEANS" ) ) {
+        // Not ideal as oceans can start later than these settings but it's at least reliably stopping before them
+        const int ocean_start_north = settings->overmap_ocean.ocean_start_north == 0 ? INT_MAX :
+                                      settings->overmap_ocean.ocean_start_north;
+        const int ocean_start_east = settings->overmap_ocean.ocean_start_east == 0 ? INT_MAX :
+                                     settings->overmap_ocean.ocean_start_east;
+        const int ocean_start_west = settings->overmap_ocean.ocean_start_west == 0 ? INT_MAX :
+                                     settings->overmap_ocean.ocean_start_west;
+        const int ocean_start_south = settings->overmap_ocean.ocean_start_south == 0 ? INT_MAX :
+                                      settings->overmap_ocean.ocean_start_south;
+        // Don't place highways over the ocean
+        if( this_om_y <= -ocean_start_north || this_om_x >= ocean_start_east ||
+            this_om_y >= ocean_start_south || this_om_x <= -ocean_start_west ) {
+            return;
+        }
+        // Check if we need to place partial highway with different intersections
+        ocean_next_north = this_om_y - 1 == -ocean_start_north;
+        ocean_next_east = this_om_x + 1 == ocean_start_east;
+        ocean_next_south = this_om_y + 1 == ocean_start_south;
+        ocean_next_west = this_om_x - 1 == -ocean_start_west;
+        if( ocean_next_north && ocean_next_east && ocean_next_south && ocean_next_west ) {
+            // This should never happen but would break everything
+            debugmsg( "Not placing highways because expecting ocean on all sides?!" );
+            return;
+        }
+    }
+    // Distance in overmaps between vertical highways ( whole overmap gap of column_seperation - 1 )
+    const int &c_seperation = settings->overmap_highway.grid_column_seperation;
+    // Distance in overmaps between horizontal highways ( whole overmap gap of row_seperation - 1 )
+    const int &r_seperation = settings->overmap_highway.grid_row_seperation;
+    // Width of segments
+    const int &segment_width = settings->overmap_highway.width_of_segments;
```

Comment:
Don't really need to store them as references. An `int` fits in a register so copying value is more efficient.
```suggestion
    // Distance in overmaps between vertical highways ( whole overmap gap of column_seperation - 1 )
    const int c_seperation = settings->overmap_highway.grid_column_seperation;
    // Distance in overmaps between horizontal highways ( whole overmap gap of row_seperation - 1 )
    const int r_seperation = settings->overmap_highway.grid_row_seperation;
    // Width of segments
    const int segment_width = settings->overmap_highway.width_of_segments;
```